src/app/qgisapp.h

```diff
@@ -2190,6 +2205,11 @@ class APP_EXPORT QgisApp : public QMainWindow, private Ui::MainWindow
     QgsFeature duplicateFeatures( QgsMapLayer *mlayer, const QgsFeature &feature );
     QgsFeature duplicateFeatureDigitized( QgsMapLayer *mlayer, const QgsFeature &feature );
 
+    //! Internal vars supporting Settings Pages function
+    QMap< QString, QString > *mOptionsPagesMap = nullptr;
```

Comment:
I wonder if you could use `std::unique_ptr` for these vars to make memory management automatic.

Reply:
+1

Reply:
Or alternatively, don't make them pointers at all, and check for empty maps as a way of detecting whether they need to be populated.